modules/alloy/src/main/ts/ephox/alloy/behaviour/coupling/CouplingState.ts

```diff
@@ -1,16 +1,27 @@
-import { Fun, Obj } from '@ephox/katamari';
+import { Fun, Obj, Optional } from '@ephox/katamari';
 
 import { AlloyComponent } from '../../api/component/ComponentApi';
 import { nuState } from '../common/BehaviourState';
 import { CouplingConfig, CouplingState } from './CouplingTypes';
 
+// Unfortunately, the Coupling APIs currently throw errors when the coupled name
+// is not recognised. This is because that if the wrong name is used, it is a
+// non-recoverable error, and the developer should be notified. However, there are
+// better ways to do this.
 const init = (): CouplingState => {
   const coupled: Record<string, AlloyComponent> = { };
 
   const getOrCreate = (component: AlloyComponent, coupleConfig: CouplingConfig, name: string): AlloyComponent => {
+    // These are the situations:
+    // 1. If we don't have any coupling configuration of any kind, then Error
+    // 2. If we already have a built coupled component, return it
+    // 3. If we haven't already built the coupled component, and we don't recognise its name,
+    // then Error
+    // 4. If we haven't already built the coupled component, but we *do* recognise its name
+    // then build it and return it.
     const available = Obj.keys(coupleConfig.others);
-    if (!available) {
-      throw new Error('Cannot find coupled component: ' + name + '. Known coupled components: ' + JSON.stringify(available, null, 2));
+    if (!available || available.length === 0) {
```

Comment:
When would `available` be nullable in this case? I think this could just be replaced entirely by `available.length === 0`?
```suggestion
    if (available.length === 0) {
```

Reply:
Sounds good.